### ISSUE TITLE
Release 1.6 candidate - CTS tests failing

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#2850]
Running CTS tests under k8s on the 1.6 branch 7ee41d1d76bfda41152887d73f986759dec155f4

Inmem:
```
Date of conformance test:          2020-04-02T13:29:15.741+0000
Number of tests that were run:     3957
Number of tests that failed:       1257
Number of tests that were skipped: 0
```
```
	id	name	description	profilePriority	conformanceStatus
0	0	Metadata sharing	The technology under test is able to share met...	MANDATORY_PROFILE	NOT_CONFORMANT
1	1	Reference copies	The technology under test is able to save, loc...	OPTIONAL_PROFILE	NOT_CONFORMANT
2	2	Metadata maintenance	The technology under test supports requests to...	OPTIONAL_PROFILE	NOT_CONFORMANT
3	3	Dynamic types	The technology under test supports changes to ...	OPTIONAL_PROFILE	UNKNOWN_STATUS
4	4	Graph queries	The technology under test supports graph-like ...	OPTIONAL_PROFILE	NOT_CONFORMANT
5	5	Historical search	The technology under test supports search for ...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
6	6	Entity proxies	The technology under test is able to store stu...	OPTIONAL_PROFILE	NOT_CONFORMANT
7	7	Soft-delete and restore	The technology under test allows an instance t...	OPTIONAL_PROFILE	NOT_CONFORMANT
8	8	Undo an update	The technology under test is able to restore a...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
9	9	Reidentify instance	The technology under test supports the command...	OPTIONAL_PROFILE	NOT_CONFORMANT
10	10	Retype instance	The technology under test supports the command...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
11	11	Rehome instance	The technology under test supports the command...	OPTIONAL_PROFILE	UNKNOWN_STATUS
12	12	Relationship search	The technology under test supports the ability...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
13	13	Entity advanced search	The technology under test supports the use of ...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
14	14	Relationship advanced search	The technology under test supports the use of ...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT

```

**Comment by planetf1** [comment]
Full output: https://ibm.box.com/s/6emy5qt3z0wt17pf1func4inac9lsqnb

**Comment by planetf1** [comment]
Identical report running under docker-compose.

**Comment by planetf1** [comment]
Graph on k8s:

```
Date of conformance test:          2020-04-02T14:36:03.993+0000
Number of tests that were run:     3955
Number of tests that failed:       1241
Number of tests that were skipped: 0
```
```
0	0	Metadata sharing	The technology under test is able to share met...	MANDATORY_PROFILE	CONFORMANT_FULL_SUPPORT
1	1	Reference copies	The technology under test is able to save, loc...	OPTIONAL_PROFILE	UNKNOWN_STATUS
2	2	Metadata maintenance	The technology under test supports requests to...	OPTIONAL_PROFILE	NOT_CONFORMANT
3	3	Dynamic types	The technology under test supports changes to ...	OPTIONAL_PROFILE	UNKNOWN_STATUS
4	4	Graph queries	The technology under test supports graph-like ...	OPTIONAL_PROFILE	UNKNOWN_STATUS
5	5	Historical search	The technology under test supports search for ...	OPTIONAL_PROFILE	NOT_CONFORMANT
6	6	Entity proxies	The technology under test is able to store stu...	OPTIONAL_PROFILE	UNKNOWN_STATUS
7	7	Soft-delete and restore	The technology under test allows an instance t...	OPTIONAL_PROFILE	NOT_CONFORMANT
8	8	Undo an update	The technology under test is able to restore a...	OPTIONAL_PROFILE	NOT_CONFORMANT
9	9	Reidentify instance	The technology under test supports the command...	OPTIONAL_PROFILE	UNKNOWN_STATUS
10	10	Retype instance	The technology under test supports the command...	OPTIONAL_PROFILE	UNKNOWN_STATUS
11	11	Rehome instance	The technology under test supports the command...	OPTIONAL_PROFILE	UNKNOWN_STATUS
12	12	Relationship search	The technology under test supports the ability...	OPTIONAL_PROFILE	UNKNOWN_STATUS
13	13	Entity advanced search	The technology under test supports the use of ...	OPTIONAL_PROFILE	UNKNOWN_STATUS
14	14	Relationship advanced search	The technology under test supports the use of ...	OPTIONAL_PROFILE	UNKNOWN_STATUS
```

**Comment by grahamwallis** [comment]
I have only looked at a subset of the results but they look very odd.
It appears that the repository content validator is not hooked up properly because the repo (in this case in-memory, but that is apparently doing the right things) is being allowed to update things like status on a ref copy and update status on a master copy to DELETED - these things are not allowed and the CTS deliberately tests that they fail. However, they are passing.

Needs more investigation. I will try and look at it some more.

**Comment by grahamwallis** [comment]
I am running a full local CTS test (against current master) on in-memory.

**Comment by grahamwallis** [comment]
Identical results with in-mem on master - currently digging into what is going on...

![image](https://user-images.githubusercontent.com/17125304/78344203-f0ded880-7593-11ea-9d98-551513197dd3.png)

**Comment by planetf1** [comment]
It's looking as if the breakage may have been introduced around the merge of 1f94801eb826463886b4a45cb60560f353404c60

**Comment by planetf1** [comment]
Still getting CTS issues on 1.6

Running under k8s.
The docker image build date is Apri 6 0923, and the last git commit is from 2020-04-03T17:14:20+0000 which is after the backport.

Will add link to full report once complete.

Reopening - I will investigate in another environment and master also

**Comment by planetf1** [comment]
User error - fix not merged yet!

**Comment by planetf1** [comment]
Code does not contain fix -- the commit for 1.6 had no code change
https://github.com/odpi/egeria/pull/2861
